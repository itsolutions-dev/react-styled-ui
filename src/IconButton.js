// @flow

import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
  getBackgroundColor,
  getBorder,
  getOpacity,
} from './utils/';

const defaultSize = '66px';
const IconButton = styled({
  component: 'button',
  style: css`
    border: ${props => getBorder(props)};
    background-color: ${props => getBackgroundColor(props)};
    opacity: ${props => getOpacity(props)};
    box-sizing: border-box;
    cursor: pointer;
    margin: 0px;
    padding: 16px;
    outline: none;
    position: relative;
    width: ${props => getSizeOrDefault(props.size, defaultSize)};
    height: ${props => getSizeOrDefault(props.size, defaultSize)};
    border-radius: 99999px;
    &:focus {
      outline: none;
    }
  `,
});

export default IconButton;
