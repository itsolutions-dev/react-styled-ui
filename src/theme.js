export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
