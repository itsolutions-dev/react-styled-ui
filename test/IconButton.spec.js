import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import IconButton from '../src/IconButton';

const SVGExample = (
  <svg viewBox="0 0 24 24" style={{ display: 'inline-block', color: 'rgba(0, 0, 0, 0.87)', fill: 'currentcolor', height: 36, width: 36, userSelect: 'none', transition: 'all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms' }}>
    <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
  </svg>);
describe('IconButton', () => {
  test('should render a IconButton', () => {
    const component = ReactTestRenderer.create(<IconButton theme={theme}>{SVGExample}</IconButton>);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a button tag [TXT]', () => {
    const component = ReactTestRenderer.create(<IconButton theme={theme}>{SVGExample}</IconButton>).toJSON();
    expect(component.type).toEqual('button');
  });

  test('should have a default style', () => {
    const component = ReactTestRenderer.create(<IconButton theme={theme}>{SVGExample}</IconButton>);
    expect(component).toHaveStyleRule('border-radius', '99999px');
    expect(component).toHaveStyleRule('background-color', '#ffffff');
    expect(component).toHaveStyleRule('opacity', '1');
    expect(component).toHaveStyleRule('width', '66px');
    expect(component).toHaveStyleRule('height', '66px');
  });
});
