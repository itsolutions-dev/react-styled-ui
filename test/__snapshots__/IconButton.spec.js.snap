// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render a IconButton 1`] = `
.c0 {
  border: #bababa;
  background-color: #ffffff;
  opacity: 1;
  box-sizing: border-box;
  cursor: pointer;
  margin: 0px;
  padding: 16px;
  outline: none;
  position: relative;
  width: 66px;
  height: 66px;
  border-radius: 99999px;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
>
  <svg
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fill": "currentcolor",
        "height": 36,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "width": 36,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
</button>
`;
